core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.ts

```diff
@@ -102,20 +119,49 @@ export class PracticeTabComponent implements OnInit {
           this.availableSubtopics[idx].getId());
       }
     }
-    const practiceSessionsUrl = this.urlInterpolationService.interpolateUrl(
-      PracticeSessionPageConstants.PRACTICE_SESSIONS_URL, {
-        topic_url_fragment: (
-          this.urlService.getTopicUrlFragmentFromLearnerUrl()),
-        classroom_url_fragment: (
-          this.urlService.getClassroomUrlFragmentFromLearnerUrl()),
-        comma_separated_subtopic_ids: selectedSubtopicIds.join(',')
-      });
+    let practiceSessionsUrl;
+    if (this.displayArea === 'topicViewer') {
+      practiceSessionsUrl = this.urlInterpolationService.interpolateUrl(
+        PracticeSessionPageConstants.PRACTICE_SESSIONS_URL, {
+          topic_url_fragment: (
+            this.urlService.getTopicUrlFragmentFromLearnerUrl()),
+          classroom_url_fragment: (
+            this.urlService.getClassroomUrlFragmentFromLearnerUrl()),
+          comma_separated_subtopic_ids: selectedSubtopicIds.join(',')
```

Comment:
Wouldn't it be possible to set `this.topicUrl` and `this.classroomUrl` in the init function, based on context(get from url or not) and not use an 'if' condition here?

Reply:
Done!